src/web/rustdoc.rs

```diff
@@ -377,18 +353,33 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         inner_path.join("/")
     };
 
+    rendering_time.step("rewrite html");
+
+    let file_content = ctry!(req, std::str::from_utf8(&file.0.content));
+    let templates = req
+        .extensions
+        .get::<super::TemplateData>()
+        .expect("missing TemplateData from the request extensions");
     // Build the page of documentation
-    RustdocPage {
-        latest_path,
-        latest_version,
-        inner_path,
-        is_latest_version,
-        rustdoc_head,
-        rustdoc_body,
-        rustdoc_body_class,
-        krate,
-    }
-    .into_response(req)
+    let ctx = ctry!(
+        req,
+        tera::Context::from_serialize(RustdocPage {
+            latest_path,
+            latest_version,
+            inner_path,
+            is_latest_version,
+            krate,
+        })
+    );
+    // Extract the head and body of the rustdoc file so that we can insert it into our own html
+    let html = ctry!(req, utils::rewrite_lol(file_content, ctx, templates));
```

Comment:
I think using a [`HtmlRewriter`](https://docs.rs/lol_html/0.2.0/lol_html/struct.HtmlRewriter.html) would be better here for two reasons, first it means we can use `MemorySettings` since `rewrite_str` currently doesn't and second it means that we can avoid parsing the file's content as utf8 since `HtmlRewriter` [takes in bytes](https://docs.rs/lol_html/0.2.0/lol_html/struct.HtmlRewriter.html#method.write)

Reply:
@Kixiron what size do you think should we use here? Previously we were guesstimating based on the file size, but this lets us configure the actual memory usage of the parser.

I'm going to go with 350 MB for now and we can change it later.

Reply:
Yah, that's something you or Pietro would know better based off of the available hardware. iirc the limits are set somewhere within our `Config` struct, if not then adding env vars would be a good path I'd think

Reply:
Those are limits for the file size, not for the memory used to parse it. I'll add an environment variable.